rpc/blockchain.go

```diff
@@ -41,10 +43,29 @@ func NewPublicBlockchainAPI(hmy *hmy.Harmony, version Version) rpc.API {
 	}
 }
 
+// ChainId returns the chain id of the chain - required by MetaMask
+func (s *PublicBlockchainService) ChainId(ctx context.Context) (interface{}, error) {
+	// Format return base on version
+	switch s.version {
+	case V1:
+		return hexutil.Uint64(s.hmy.ChainID), nil
+	case V2:
+		return s.hmy.ChainID, nil
+	case Eth:
+		shardID, err := nodeconfig.GetDefaultConfig().ShardIDFromConsensusKey()
+		if err != nil {
+			return nil, err
+		}
+		return hexutil.Uint64(params.EthMainnetChainID.Uint64() + uint64(shardID)), nil
```

Comment:
You can directly use Eth<network>ChainID since it's updated to the correct one based on ShardID at the start of the node with `params.UpdateEthChainIDByShard(nodeConfig.ShardID)`